container-runtime/src/Adapters/Systems/SystemForEmbassy/index.ts

```diff
@@ -179,7 +180,38 @@ const matchProperties = object({
   version: literal(2),
   data: matchPackageProperties,
 })
+const matchResult = object({
+  result: any,
+})
+const matchError = object({
+  error: string,
+})
+const matchErrorCode = object<{
+  "error-code": [number, string] | readonly [number, string]
+}>({
+  "error-code": tuple(number, string),
+})
 
+const assertNever = (
+  x: never,
+  message?: "Not expecting to get here",
+): never => {
+  throw new Error("")
```

Comment:
Maybe include an actual error message in for better debugging in case it's tripped